components/idp-mgt/org.wso2.carbon.idp.mgt/src/main/java/org/wso2/carbon/idp/mgt/dao/IdPManagementFacade.java

```diff
@@ -0,0 +1,427 @@
+    }
+
+    public void renameTenantRole(String newRoleName, String oldRoleName, int tenantId, String tenantDomain)
+            throws IdentityProviderManagementException {
+
+        dao.renameTenantRole(newRoleName, oldRoleName, tenantId, tenantDomain);
+    }
+
+    public void renameClaimURI(String newClaimURI, String oldClaimURI, int tenantId, String tenantDomain)
+            throws IdentityProviderManagementException {
+
+        dao.renameClaimURI(newClaimURI, oldClaimURI, tenantId, tenantDomain);
+    }
+
+    public boolean isIdPAvailableForAuthenticatorProperty(String authenticatorName, String propertyName,
+                                                          String idPEntityId, int tenantId)
+            throws IdentityProviderManagementException {
+
+        return dao.isIdPAvailableForAuthenticatorProperty(authenticatorName, propertyName, idPEntityId, tenantId);
+    }
+
+    public ConnectedAppsResult getConnectedApplications(String resourceId, int limit, int offset)
+            throws IdentityProviderManagementException {
+
+        return dao.getConnectedApplications(resourceId, limit, offset);
+    }
+
+    public ConnectedAppsResult getConnectedAppsOfLocalAuthenticator(String authenticatorId, int tenantId,
+                                                                    Integer limit, Integer offset)
+            throws IdentityProviderManagementException {
+
+        return dao.getConnectedAppsOfLocalAuthenticator(authenticatorId, tenantId, limit, offset);
+    }
+
+    public String getIdPNameByMetadataProperty(Connection dbConnection, String property, String value, int tenantId)
+            throws IdentityProviderManagementException {
+
+        return dao.getIdPNameByMetadataProperty(dbConnection, property, value, tenantId);
+    }
+
+    public boolean isClaimReferredByAnyIdp(Connection dbConnection, String claimUri, int tenantId)
+            throws IdentityProviderManagementException {
+
+        return dao.isClaimReferredByAnyIdp(dbConnection, claimUri, tenantId);
+    }
+
+    public Map<String, String> getIdPNamesById(int tenantId, Set<String> idpIds)
+            throws IdentityProviderManagementException {
+
+        return dao.getIdPNamesById(tenantId, idpIds);
+    }
+
+    public List<IdPGroup> getIdPGroupsByIds(List<String> idpGroupIds, int tenantId)
+            throws IdentityProviderManagementException {
+
+        return dao.getIdPGroupsByIds(idpGroupIds, tenantId);
+    }
+
+    public List<FederatedAuthenticatorConfig> getAllUserDefinedFederatedAuthenticators(int tenantId)
+            throws IdentityProviderManagementException {
+
+        List<FederatedAuthenticatorConfig> configList = dao.getAllUserDefinedFederatedAuthenticators(tenantId);
+        for (FederatedAuthenticatorConfig config : configList) {
+            endpointConfigurationManager.resolveEndpointConfig(config, IdentityTenantUtil.getTenantDomain(tenantId));
+        }
+        return configList;
```

Comment:
ideally we should not loop here as well and it should be possible to resolve the actions for the given type.

Let's also track this as an improvement as the impact can be in management layer only not it execution layer

Reply:
This will be tracked with issue: https://github.com/wso2/product-is/issues/21783